src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -579,6 +609,15 @@ public static void writeTo(ScriptType scriptType, StreamOutput out) throws IOExc
                 out.writeVInt(INLINE_VAL); //Default to inline
             }
         }
+
+        @Override
+        public String toString() {
+            //ScriptModes looks for 'dynamic' rather than 'inline' when reading settings, we should rename but we want to keep bw comp for now.
+            if (this == INLINE) {
+                return "dynamic";
```

Comment:
@clintongormley we use `inline` instead of `dynamic` to refer to dynamic scripts... I added this special case to convert ScriptType.INLINE to the `dynamic` string in a bw comp manner. At some point we will have to choose for one of the two. Wondering which one we prefer: either use inline in settings here or rename ScriptType.INLINE to DYNAMIC in the future (master, it's a breaking change).